### ISSUE TITLE
Rancher never loads "cluster groups" when managing Fleet inside local cluster

### ISSUE THREAD

**Issue by gaktive** [rancher/dashboard#8540]
Internal reference: SURE-6063
Reported in 2.6.9 & 2.7.0

Issue description:
A user noticed that if you are in the Rancher UI and you go to "local" cluster > More Resources > Fleet > Cluster Groups > then select a cluster group, it never loads. It just stays "loading..." no matter how long you leave it. So far, this is in testing phase of Fleet so there's no impact.

Troubleshooting steps:
The user had a look at the browser tools and the console tab says:

```
TypeError: Cannot read properties of undefined (reading 'clusters')
    at t.get (fleet.cattle.io.clustergroup.js:25:38)
    at f.show (FleetSummaryGraph.vue:23:1)
    at Tn.get (vue.runtime.esm.js:4495:25)
    at Tn.evaluate (vue.runtime.esm.js:4597:21)
    at f.show (vue.runtime.esm.js:4851:17)
    at f.<anonymous> (FleetSummaryGraph.vue?331a:1:106)
    at t._render (vue.runtime.esm.js:3569:22)
    at f.r (vue.runtime.esm.js:4081:21)
    at Tn.get (vue.runtime.esm.js:4495:25)
    at new Tn (vue.runtime.esm.js:4484:12)
```

Repro steps:

step 1 Go to local cluster
step 2 Select "more resources"
step 3 Select Fleet
step 4 select cluster groups
step 5 select a cluster groups
step 6 cluster group never loads

Workaround:
Using the "Continuous Delivery" option in the main hamburger menu > Cluster groups > select a Cluster group. It loads just fine

Actual behavior:
Doesn't load when managing Fleet from the local cluster

Expected behavior:
Should work from both places

**Comment by gaktive** [comment]
@aalves08 indicated this works in `master` and `2.7.2-rc7`. So goal here is to do a quick repro on 2.7.0 and verify that yes, this is fixed.

**Comment by aalves08** [comment]
@gaktive I found the issue on `2.7.2`. Yesterday I don't know what I was running, but wasn't hitting it. Created PRs to fix both for `master` and `release-2.7.2`.

**Comment by aalves08** [comment]
PR for master: https://github.com/rancher/dashboard/pull/8641
PR for backporting to 2.7.2: https://github.com/rancher/dashboard/pull/8642

@gaktive

**Comment by markusewalker** [comment]
Validated this is addressed on `v2.7-head` https://github.com/rancher/rancher/commit/1712bf9f7116e25711fd76b99d7ce5672c2d4a23. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/1712bf9f7116e25711fd76b99d7ce5672c2d4a23
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As an admin user, attempt to load the `default` Fleet cluster group and validate it loads in Cluster Management and Continuous Delivery|:white_check_mark:|
|2|As an admin user, attempt to load a created Fleet cluster group and validate it loads in Cluster Management and Continuous Delivery|:white_check_mark:|
|3|As an admin user, create multiple Fleet cluster groups and validate all of them load in Cluster Management and Continuous Delivery|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. As an admin user, navigated to top-left menu > local > More Resources > Fleet > Cluster Groups.
2. Validated that I am able to load the `default` cluster group.
3. Navigated to Continuous Delivery > Cluster Groups.
4. Validated that I am able to load the `default` cluster group.
__________
**Scenario 2**
1. Repeated scenario 1, but using a new cluster group that I created.
2. Validated that I could access the custom cluster group in Cluster Management and Continuous Delivery.
3. Validated that I could successfully delete the cluster group.
_________
**Scenario 3**
1. Repeated scenario 1, but created 5 cluster groups instead of just 1 as I did in scenario 2.
2. Validated that I could access all of the custom cluster group in Cluster Management and Continuous Delivery.
3. Validated that I could successfully delete all of the cluster groups.
